server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/DeletedMessagesVaultTest.java

```diff
@@ -610,13 +612,17 @@ public void vaultExportShouldResponseIdempotentSideEffect() throws Exception {
     }
 
     @Test
-    public void vaultPurgeShouldDeleteExpiredMessageThenExportWithNoEntry() throws Exception {
-        Instant now = Instant.now();
+    public void vaultPurgeShouldMakeExportProduceEmptyZipWhenAllMessagesAreExpired() throws Exception {
         when(clock.instant())
-            .thenReturn(now, now.plus(366, ChronoUnit.DAYS));
+            .thenReturn(NOW, NOW, NOW);
 
         bartSendMessageToHomer();
-        WAIT_TWO_MINUTES.until(() -> listMessageIdsForAccount(homerAccessToken).size() == 1);
+        bartSendMessageToHomer();
+        bartSendMessageToHomer();
+        WAIT_TWO_MINUTES.until(() -> listMessageIdsForAccount(homerAccessToken).size() == 3);
+
+        when(clock.instant())
+            .thenReturn(NOW, ONE_DAY_AFTER_ONE_YEAR_EXPIRATION);
```

Comment:
Does this make all 3 messages expired? When I look at the code overall I doubt it... but I could be wrong !